private/buf/cmd/buf/buf_test.go

```diff
@@ -3596,18 +3595,65 @@ func TestConvertWithImage(t *testing.T) {
 		assert.JSONEq(t, `{"one":"55"}`, stdout.String())
 	})
 
-	t.Run("no stdin input", func(t *testing.T) {
+	t.Run("no stdin input from binpb", func(t *testing.T) {
+		t.Parallel()
+		testRun(
+			t,
+			0,
+			nil,
+			nil,
+			"convert",
+			filepath.Join(tempDir, "image.binpb"),
+			"--type",
+			"buf.Foo",
+			"--from=-#format=binpb",
+			"--to=-#format=txtpb",
+		)
+	})
+	t.Run("no stdin input from txtpb", func(t *testing.T) {
+		t.Parallel()
+		testRun(
+			t,
+			0,
+			nil,
+			nil,
+			"convert",
+			filepath.Join(tempDir, "image.binpb"),
+			"--type",
+			"buf.Foo",
+			"--from=-#format=txtpb",
+			"--to=-#format=binpb",
+		)
+	})
+	t.Run("no stdin input from yaml", func(t *testing.T) {
+		t.Parallel()
+		testRun(
+			t,
+			0,
+			nil,
+			nil,
+			"convert",
+			filepath.Join(tempDir, "image.binpb"),
+			"--type",
+			"buf.Foo",
+			"--from=-#format=yaml",
+			"--to=-#format=binpb",
+		)
+	})
+	t.Run("no stdin input from json", func(t *testing.T) {
 		t.Parallel()
 		testRunStdoutStderrNoWarn(
 			t,
 			nil,
 			1,
 			"",
-			`Failure: --from: length of data read from "-" was zero`,
+			`Failure: --from: proto: syntax error (line 1:1): unexpected token`,
```

Comment:
Not a big fan of the error message returned from the json unmarshaler.
We could wrap the error:

https://github.com/bufbuild/buf/blob/4d05f30e35c0fbee2c384878b55d6adac4477fb6/private/pkg/protoencoding/json_unmarshaler.go#L45

```
func (m *jsonUnmarshaler) Unmarshal(data []byte, message proto.Message) error {
	options := protojson.UnmarshalOptions{
		Resolver:       m.resolver,
		DiscardUnknown: !m.disallowUnknown,
	}
  if err := options.Unmarshal(data, message); err != nil {
    return fmt.Errorf("json unmarshal: %w", err)
  }
  return nil
}
```